src/main/java/edu/uwaterloo/ece658/service/DataTransmissionService.java

```diff
@@ -15,19 +27,124 @@
 @Stateless
 public class DataTransmissionService {
 
-//    public String upload(String pathToImage) {
-//        return null;
-//    }
-//
-//    public boolean updateS3key(String S3key, List<String> tags) {
-//        return false;
-//    }
-//
-//    public String download(String S3key) {
-//        return null;
-//    }
-//
-//    public boolean deleteImage(String S3key) {
-//        return false;
-//    }
+    @EJB
+    private UserFacade userFacade;
+
+    @EJB
+    private SubscriptionService subscriptionService;
+
+    @EJB
+    private PhotoFacade photoFacade;
+    
+    @EJB
+    private TagFacade tagFacade;
+
+    @EJB
+    private S3Util s3Util;
+
+    public URL getUploadURL(String s3Key) {

```

Comment:
Let's move getUploadURL and getDownloadURL methods into a new Service called S3Service. You can simply rename S3Util to S3Service, I think.

Reply:
Sure